### PR TITLE
feat(1415): Add items to data schema for external repo validation

### PR DESCRIPTION
**Pull request by supra08**
## Context

Fixes screwdriver-cd/screwdriver#1415

## Objective

This PR adds items to the data-schema for handling new config and updated functions for external repo notifications

## References

screwdriver-cd/screwdriver#1415

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config/base.js [hunk 1]
```diff
@@ -39,6 +39,12 @@ const SCHEMA_CHILD_PIPELINES = Joi.object()
         startAll: Joi.boolean()
     })
     .unknown(false);
+
+const SCHEMA_SUBSCRIBE = Joi.object()
+    .keys({
+        scmUrls: Joi.array().items(Joi.object())
```
Comment (tkyi) [inline config/base.js:45]:
```suggestion
        scmUrls: SCHEMA_SCM_URLS.required()
```

could reuse predefined scm urls

Comment (tkyi) [reply config/base.js:45]: or if the regex is not the same, might be better to use a different key

Comment (supra08) [reply config/base.js:45]: Yeah, I should probably go with the different key.

File: models/pipeline.js [hunk 2]
```diff
@@ -77,7 +77,16 @@ const MODEL = {
...
+    parameters: Parameters.parameters,
+
+    // This field is kept for making the pipelines queryable.
+    subscribedScmUrls: Joi.array().items(Joi.string().regex(Regex.SCM_URI))
+        .description('List of subscribed scm urls'),
+
+    subscribedScmUrlsWithActions: Joi.array().items(Joi.object().keys({
+        scmUri: Regex.SCM_URI,
+        actions: Joi.array().items(Joi.string())
+    })).description('List of subscribed scm urls paired with actions')
```
Comment (tkyi) [inline models/pipeline.js:89]: Just wondering why we need both
is it possible to condense?

File: models/build.js [hunk 3]
```diff
@@ -66,6 +66,11 @@ const MODEL = {
         .description('SHA this project was built on')
         .example('ccc49349d3cffbd12ea9e3d41521480b4aa5de5f'),
 
+    subscribedConfigSha: Joi
```
Comment (kumada626) [inline models/build.js:69]: Do we need to add migration file to add this column on builds?
https://github.com/screwdriver-cd/data-schema/tree/master/migrations

File: models/pipeline.js [hunk 4]
```diff
@@ -100,7 +105,7 @@ module.exports = {
     ], [
         'workflowGraph', 'scmRepo', 'annotations', 'lastEventId',
         'configPipelineId', 'childPipelines', 'name', 'prChain',
-        'parameters'
+        'parameters', 'subscribedScmUrlsWithActions', 'subscribedScmUrls'
```
Comment (tkyi) [inline models/pipeline.js:108]:
```suggestion
        'parameters', 'subscribedScmUrlsWithActions'
```

File: migrations/20190919-initdb-builds.js [hunk 5]
```diff
@@ -71,6 +71,9 @@ module.exports = {
                 },
                 stats: {
                     type: Sequelize.TEXT
+                },
+                subscribedConfigSha: {
```
Comment (kumada626) [inline migrations/20190919-initdb-builds.js:75]: Migration files should not be upgraded but added like this file.
https://github.com/screwdriver-cd/data-schema/blob/master/migrations/20200204-add-column_templateId_to_builds.js

**Review by jithine — APPROVED** [review]

**Pull request merged**